backend/src/api/routes/surveys.py

```diff
@@ -36,7 +37,9 @@ async def get_surveys_for_user(
         raise HTTPException(status_code=400, detail="User not found")
 
     user_id = user_crud.get_user_by_email(user.user_email, session).id
-    user_surveys = survey_crud.get_all_surveys_for_user(user_id, session)
+    user_surveys = survey_crud.get_not_deleted_surveys_for_user(
```

Comment:
Do not create a new get_not_deleted_surveys_for_user method. Instead, reimplement get_all_surveys_for_user to only return non-deleted surveys. We will never want to return deleted surveys, and only the crud component should be aware of how the deletion is handled. The crud component should hide details like this.